Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -185,7 +181,12 @@ public struct BSONDocument {
             BSONDocumentIterator.find(key: key, in: self)?.value
         }
         set {
-            self.set(key: key, to: newValue)
+            // We have no choice but to crash here.
```

Comment:
Similarly, we are not allowed to throw from subscript setters/getters. (well, we can as of Swift 5.5, but we support Swift 5.1+ right now). I opened SWIFT-1357 about starting to throw from the subscript in our next major version